interview-questions.md

```diff
@@ -8,44 +8,78 @@ Then, research the question to expand on your answer. Even if you feel you have
+Hierarchical- Has a tree like structure that is a very rigid structure and realtionships are pre-defined such as the folder system in our computers.
+Network- Has a tree like structure that is a bit more flexible than hierarchical the children can have more that one realtion to a parent.
+Object oriented- Unstructures or semi-structured but uses key-values in some cases.
+Depending on the type opf data that you are storing or industry that your are in one of the databases my be better suited that the others for your specific needs. In web development the relational is the one most widely used.
+
+https://www.nibusinessinfo.co.uk/content/types-database-system
 
 4. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: primary key is the unique identifying item for the instance of the class.
+
+Researched answer: Similar to arrays that use a index system to track the values of the data that it is storing that is one way we can think of a primary key to a database table. It is a unique identifier that does not change over time it is consistent and it give the user the abilty to retrive data and manipulate data effieciently and with out adding it to the worng column. One of the other functions is that the primary key is used to reference relationships of tables inside of the database. Just like a drivers license number or a social security number for an adult those numbers are unique to eah person and are not repreated think about someone having your sasme social security number that would create a significant amount of confusing when applying for a new job or when you go to the hospital when your information is pulled and wanting to be verified there would be multiple people under that same number. This could be applied to when calling information form a database and you are trying to call a specific set of data but the return has many entries that come back to you making harder to decipher and can cause consfusion.
 
-Researched answer:
+https://www.simplilearn.com/tutorials/sql-tutorial/primary-key-in-sql#:~:text=The%20primary%20key%20in%20SQL,unique%20values%20for%20each%20row.
 
 5. What are the HTTP verbs? What is each verb's corresponding CRUD action?
 
-Your answer:
+Your answer: It is list of commands that communicates from the front-end of an application to the backend.
```

Comment:
Give some more elaboration on the 'your answers' sections! Treat it like an interview!